Stack/Opc.Ua.Core/Stack/Server/ServerBase.cs

```diff
@@ -778,14 +815,17 @@ protected virtual void OnCertificateUpdate(object sender, CertificateUpdateEvent
                         settings.ServerCertificateChain = this.InstanceCertificateChain;
                     }
 
-                    ITransportListener listener = new Opc.Ua.Bindings.UaTcpChannelListener();
+                    ITransportListener listener = new Opc.Ua.Bindings.TcpListener();
 
                     listener.Open(
                        uri.Uri,
                        settings,
                        GetEndpointInstance(this));
 
                     TransportListeners.Add(listener);
+
+                    listener.ConnectionWaiting += OnConnectionWaiting;
+                    listener.ConnectionStatusChanged += OnConnectionStatusChanged;
```

Comment:
Unregistration from events is missing

Reply:
this needs some investigation...

---
In reply to: [432434590](https://github.com/OPCFoundation/UA-.NETStandard/pull/998#discussion_r432434590) [](ancestors = 432434590)

Reply:
clear events on listener dispose.

---
In reply to: [433798419](https://github.com/OPCFoundation/UA-.NETStandard/pull/998#discussion_r433798419) [](ancestors = 433798419,432434590)